units/recalculate_change_points.go

```diff
@@ -0,0 +1,243 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/cedar/perf"
+	"github.com/mongodb/grip"
+	"sort"
+	"time"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/util"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+)
+
+type recalculateChangePointsJob struct {
+	*job.Base      `bson:"metadata" json:"metadata" yaml:"metadata"`
+	env            cedar.Environment
+	metricGrouping MetricGrouping
+}
+
+func init() {
+	registry.AddJobType("recalculate-change-points", func() amboy.Job { return makeChangePointsJob() })
+}
+
+func makeChangePointsJob() *recalculateChangePointsJob {
+	j := &recalculateChangePointsJob{
+		Base: &job.Base{
+			JobType: amboy.JobType{
+				Name:    "recalculate-change-points",
+				Version: 1,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func NewRecalculateChangePointsJob(metricGrouping MetricGrouping) amboy.Job {
+	j := makeChangePointsJob()
+	// Every ten minutes at most
+	timestamp := util.RoundPartOfHour(10)
+	j.SetID(fmt.Sprintf("%s.%s.%s.%s.%s.%s", j.JobType.Name, metricGrouping.Id.Project, metricGrouping.Id.Variant, metricGrouping.Id.Task, metricGrouping.Id.Test, timestamp))
+	j.metricGrouping = metricGrouping
+	return j
+}
+
+type MetricGrouping struct {
+	Id struct {
+		Project string `bson:"project"`
+		Variant string `bson:"variant"`
+		Task    string `bson:"task"`
+		Test    string `bson:"test"`
+	} `bson:"_id"`
+}
+
+func makeSeriesFilter(grouping MetricGrouping) bson.M {
+	return bson.M{
+		"info.project":   grouping.Id.Project,
+		"info.variant":   grouping.Id.Variant,
+		"info.task_name": grouping.Id.Task,
+		"info.test_name": grouping.Id.Test,
+	}
+
+}
+
+func makeTimeSeriesPipeline(grouping MetricGrouping) []bson.M {
```

Comment:
these queries should be in the model package. the units package should only have the business logic.